js/views/modals/purchase/Shipping.js

```diff
@@ -0,0 +1,43 @@
+import '../../../lib/select2';
+import app from '../../../app';
+import loadTemplate from '../../../utils/loadTemplate';
+import BaseModal from '../BaseModal';
+import Listing from '../../../models/listing/Listing';
+
+export default class extends BaseModal {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    if (!this.model || !(this.model instanceof Listing)) {
+      throw new Error('Please provide a listing model');
+    }
+
+    this.listenTo(app.settings.get('shippingAddresses'), 'update',
+      (cl, updateOpts) => {
+        if (updateOpts.changes.added.length ||
+          updateOpts.changes.removed.length) {
```

Comment:
I don't think I will call save on the order model, since the server returns data that shouldn't go into the order model. 

But I will be using the order model to render the receipt (right side bar with the prices), so it seemed like it was worth it to make it a model, so it's easy to rerender the receipt on changes etc.